expression/builtin_math.go

```diff
@@ -1074,19 +1074,33 @@ type piFunctionClass struct {
 }
 
 func (c *piFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinPISig{newBaseBuiltinFunc(args, ctx)}
-	return sig.setSelf(sig), errors.Trace(c.verifyArgs(args))
+	var (
+		bf  baseBuiltinFunc
+		sig builtinFunc
+		err error
+	)
+
+	if err = c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	if bf, err = newBaseBuiltinFuncWithTp(args, ctx, tpReal); err != nil {
+		return nil, errors.Trace(err)
+	}
+
```

Comment:
I have updated and please take a look. 

Before your comments, I found that `tidb` returns the float64 data without caring the retType. It seems that there is only one case we need to care retType for float64, which is PI() function.... It is pretty weird...

So, I tried to handle this case. It seems that we can only handle precision of float64 data after Executor.Next() call. Since there are a lot of instances implementing `Next()`, it must be breaking change. That is why I asked precision thing again and again. Sorry about that.